### PR TITLE
#168 cacheable: add mutex update cache

### PR DESCRIPTION
**Pull request by sejust**

### CODE REVIEW

File: cacheable/README.md [hunk 1]
```diff
@@ -118,6 +120,13 @@ Create a `LRU` object, all items will be cached in it.
 
     -   `False`: return reference of cached item
 
+-   `mutex_update`: mutex updating the cache if two or more callers
+    simultaneously update the item. Default is `False`.
+
+    -   `True`: mutex update
+
+    -   `False`: synchronously update
```
Comment (drmingdrmer) [inline cacheable/README.md]: concurrently update?

File: cacheable/README.md [hunk 2]
```diff
@@ -118,6 +120,13 @@ Create a `LRU` object, all items will be cached in it.
 
     -   `False`: return reference of cached item
 
+-   `mutex_update`: mutex updating the cache if two or more callers
```
Comment (drmingdrmer) [inline cacheable/README.md]: allows only one thread to update the cache?

Comment (sejust) [reply cacheable/README.md]: 1

File: cacheable/README.md (hunk 1 above)
Comment (sejust) [reply cacheable/README.md]: 1

**Pull request merged**